test_utils/utils.go

```diff
@@ -16,13 +17,58 @@ import (
 
 	"github.com/containernetworking/cni/pkg/ns"
 	"github.com/containernetworking/cni/pkg/types"
+	etcdclient "github.com/coreos/etcd/client"
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega/gexec"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	calnet "github.com/projectcalico/libcalico-go/lib/net"
 	"github.com/vishvananda/netlink"
 )
 
-func PreCreatePool(pool string) string {
-	return Cmd("dist/calicoctl pool add " + pool)
+var kapi etcdclient.KeysAPI
+
+func init() {
+
+	cfg := etcdclient.Config{Endpoints: []string{"http://127.0.0.1:2379"}}
+	c, _ := etcdclient.New(cfg)
+	kapi = etcdclient.NewKeysAPI(c)
+}
+
+// Delete everything under /calico from etcd
+func WipeEtcd() {
+	_, err := kapi.Delete(context.Background(), "/calico", &etcdclient.DeleteOptions{Dir: true, Recursive: true})
+	if err != nil && !etcdclient.IsKeyNotFound(err) {
+		panic(err)
+
+	}
+
+}
+
+func PreCreatePool(pool string) {
+	// Load the client config from the current environment.
+	clientConfig, err := client.LoadClientConfig("")
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a new client.
+	calicoClient, err := client.New(*clientConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	_, ipNet, err := calnet.ParseCIDR(pool)
```

Comment:
We also have a few "hokey" etcd methods that used to use etcdctl to clean etcd.  Might have been nice to update the test utilities.